server/log-collector/pkg/services/logcollector/server.go

```diff
@@ -372,14 +367,27 @@ func (s *Server) GetLogs(request *protologcollector.GetLogsRequest, responseStre
 		return nil
 	}
 
+	var endSize = currentLogFileSize
+
+	// if request size is positive, read only the requested size, add the offset to ensure we read
+	// all requested logs
+	if request.Size > 0 {
+		endSize = request.Size + request.Offset
+	}
```

Comment:
not sure I understand why you add the offset here to the size - we use the offset in `readAt` later on, so the size of the read logs doesn't include the offset bytes...

Reply:
u start to read at offset, and you read until size+offset
if you dont add offset, then you start to read at offset and read until size
so basically, for the latter, if you want to read 100 bytes with offset 10, you essentially read 90
see that getChunkSize uses the endsize to calculate the leftToRead in a new way